app/src/main/java/org/sopt/sample/src/login/SignInActivity.kt

```diff
@@ -0,0 +1,105 @@
+package org.sopt.sample.src.login
+
+import android.app.Activity
+import android.content.Intent
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.activity.result.contract.ActivityResultContracts
+import org.sopt.sample.R
+import org.sopt.sample.config.ApplicationClass
+import org.sopt.sample.config.BaseActivity
+import org.sopt.sample.databinding.ActivitySigninBinding
+import org.sopt.sample.src.HomeActivity
+
+
+class SignInActivity : BaseActivity<ActivitySigninBinding>(ActivitySigninBinding::inflate) {
+    private lateinit var resultLauncher: ActivityResultLauncher<Intent> //회원가입 정보를 받아오기 위한 launcher
+    private var idFromSignup: String? = null
+    private var pwFromSignup: String? = null
+    private var mbtiFromSignup: String? = null
+    private lateinit var editor:SharedPreferences.Editor
+    override fun onCreate(savedInstanceState: Bundle?) {
+
+        super.onCreate(savedInstanceState)
+        editor = ApplicationClass.sSharedPreferences.edit()
+
+        autoLogin()//자동 로그인
+        setResultSignUp() //회원가입 콜백
+        signIn() //로그인
+        signUp() //회원가입
+
+    }
+
+    //Signup Callback
+    private fun setResultSignUp() {
+        resultLauncher =
+            registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+                if (result.resultCode == Activity.RESULT_OK) {
+                    idFromSignup = result.data?.getStringExtra("id").toString()
+                    pwFromSignup = result.data?.getStringExtra("pw").toString()
+                    mbtiFromSignup = result.data?.getStringExtra("mbti").toString()
```

Comment:
아래와 같이 작성해볼 수도 있겠습니다 :) Indentation 깊이가 깊어지기 때문에 반드시 아래 코드가 좋은 예시라고는 할 순 없겠습니다! (+ 가입정보를 담는 data class를 만들어서 객체를 전달하는 방법도 있습니다)
```suggestion
                    result.data?.let {
                        idFromSignup = it.getStringExtra("id")
                        pwFromSignup = it.getStringExtra("pw")
                        mbtiFromSignup = it.getStringExtra("mbti")
                    }
```
`toString()`을 제거한 이유는 다음과 같습니다!
`result.data?.getStringExtra("id")`는 String? 타입으로 값을 가져오지 못할 때 null값을 가져올 것입니다! null값을 가져온다면 `toString()`에 의해 null이아닌 문자열 "null"로 초기화될 것입니다~

<img width="821" alt="image" src="https://user-images.githubusercontent.com/48701368/194472769-4a4f1fd3-796e-4a4e-995e-45304a41b07e.png">